src/App.tsx

```diff
@@ -15,36 +16,95 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
-export const App: React.FC = () => {
+enum SortMethod {
+  Alphabet = 'alphabet',
+  Length = 'length',
+  Default = '',
+}
+
+const prepareGoods = (
+  sortMethod: SortMethod,
+  isListReversed: boolean,
+): string[] => {
+  const goodsCopy = [...goodsFromServer];
+
+  if (sortMethod) {
+    goodsCopy.sort((a, b) => {
+      switch (sortMethod) {
+        case SortMethod.Alphabet:
+          return a.localeCompare(b);
+        case SortMethod.Length:
+          return a.length - b.length;
+        default:
+          return 0;
+      }
+    });
+  }
+
+  if (isListReversed) {
+    goodsCopy.reverse();
+  }
+
+  return goodsCopy;
+};
+
+export const App = () => {
+  const [sortMethod, setSortMethod] = useState<SortMethod>(SortMethod.Default);
+  const [isListReversed, setIsListReversed] = useState(false);
+
+  const resetAll = () => {
+    setSortMethod(SortMethod.Default);
+    setIsListReversed(false);
+  };
+
+  const isResetButtonVisible = sortMethod || isListReversed;
+
+  const goodsList = prepareGoods(sortMethod, isListReversed);
+
   return (
     <div className="section content">
       <div className="buttons">
-        <button type="button" className="button is-info is-light">
+        <button
+          onClick={() => setSortMethod(SortMethod.Alphabet)}
+          type="button"
+          className={`button is-info ${sortMethod !== SortMethod.Alphabet && 'is-light'}`}
+        >
           Sort alphabetically
         </button>
 
-        <button type="button" className="button is-success is-light">
+        <button
+          onClick={() => setSortMethod(SortMethod.Length)}
+          type="button"
+          className={`button is-success ${sortMethod !== SortMethod.Length && 'is-light'}`}
+        >
           Sort by length
         </button>
 
-        <button type="button" className="button is-warning is-light">
+        <button
+          onClick={() => setIsListReversed(!isListReversed)}
```

Comment:
```suggestion
          onClick={() => setIsListReversed(prev => !prev)}
```
If you need to set a state depending on the previous one use functional setState.